components/eamxx/src/share/io/scorpio_input.cpp

```diff
@@ -9,6 +9,10 @@
 namespace scream
 {
 
+AtmosphereInput::AtmosphereInput()
```

Comment:
Looks like you added this empty ctor b/c you added a member of type `AtmosphereInput` to `Nudging`. Instead of adding an interface just for the sake of that class, I would have `Nudging` store a `std::shared_ptr<AtmosphereInput>`, which you can then init once you have all the data, and I would remove this empty ctor.

Reply:
Thanks for the input. This is a better solution. I put std::shared_ptr<AtmosphereInput> data_input; 
in the Nudging header file and then these lines where the data is filled. I believe this is what you were suggesting.
 AtmosphereInput data_input0(m_comm,data_in_params);
  data_input0.init(grid_l,host_views,layouts);
  data_input=std::make_shared<AtmosphereInput>(data_input0);

Reply:
I also removed the empty ctor